src/main/java/graphql/execution/Execution.java

```diff
@@ -47,8 +50,17 @@ public Execution(ExecutionStrategy queryStrategy, ExecutionStrategy mutationStra
 
     public CompletableFuture<ExecutionResult> execute(Document document, GraphQLSchema graphQLSchema, ExecutionId executionId, ExecutionInput executionInput, InstrumentationState instrumentationState) {
 
```

Comment:
Previously the operation node extraction happened inside ExecutionContextBuilder.build()

I have pulled this out because I want to refactor more in a later PR to allow "business rule validation of variable input" and hence we need the operation (and variable defs) before we build the execution context.

Some of the attributes changed because they were only passed to allow this parsing.  By moving it here we dont need to pass them on any more.  For example Document.  Its only used for the getOperation say.